### ISSUE TITLE
Bottlerocket  bootstrap_extra_args are ignored for EKS submodule eks-managed-node-group

### ISSUE THREAD

**Issue by youwalther65** [terraform-aws-modules/terraform-aws-eks#2290]
## Description

When using the submodule eks-managed-node-group to create a stand-alone Bottlerocket outside modeul EKS bootstrap_extra_args are not merged into used data and Bottlerocket uses MNG defaults ignoring the extra values provided

## Versions

- Module version [Required]:
EKS module 18.30.2

- Terraform version:
1.3.4

- Provider version(s):
AWS 4.37.0

## Reproduction Code [Required]

```yaml
# See: https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/eks-managed-node-group
module "eks_managed_bottlerocket_node_group" {
  source = "terraform-aws-modules/eks/aws//modules/eks-managed-node-group"
  version = ">= 18.30.2"
...
  ami_type = "BOTTLEROCKET_x86_64"
  platform = "bottlerocket"

  bootstrap_extra_args = <<-EOT
  # extra args added
  [settings.kernel]
  lockdown = "integrity"

  [settings.host-containers.admin]
  enabled = false

  [settings.kubernetes.node-labels]
  "label1" = "waltju1"
  "label2" = "waltju2"
  EOT
...
```

## Expected behavior

extra args are reflected in user data and as node labels

## Actual behavior

default MNG userdata for Bottlerocket and bootstrap_extra_args are ignored

**Comment by youwalther65** [comment]
I need help here. I really followed the examples from the docs [here](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/eks-managed-node-group).
I tried using an approach with custom Bottlerocket AMI but it doesn't work as wlel and nodes don't even join the cluster:

module "eks_managed_bottlerocket_custom_node_group" {
  source = "terraform-aws-modules/eks/aws//modules/eks-managed-node-group"
  version = ">= 18.30.2"

  count = var.enable_br_cmng ? 1 : 0

  name = "${local.name}-cbr-mng"

#  cluster_name    = local.name
  cluster_name    = module.eks.cluster_id
  cluster_version = local.cluster_version

  desired_size = 1
  max_size     = 1
  min_size     = 0

  vpc_id     = module.vpc.vpc_id
  subnet_ids = module.vpc.private_subnets

  // The following variables are necessary if you decide to use the module outside of the parent EKS module context.
  // Without it, the security groups of the nodes are empty and thus won't join the cluster.
  // https://github.com/terraform-aws-modules/terraform-aws-eks/blob/cc3fe0590b6d7c8bf2a1690c5331fa92527efbe4/examples/complete/main.tf#L250-L265
  cluster_primary_security_group_id = module.eks.cluster_primary_security_group_id
  vpc_security_group_ids  = [
    module.eks.node_security_group_id,
    aws_security_group.additional.id,
  ]

  instance_types = ["t3.medium"]
  capacity_type  = "SPOT"

  ebs_optimized          = true
  enable_monitoring      = true

  block_device_mappings = {
    xvda = {
      device_name = "/dev/xvda"
      ebs = {
        volume_size           = 60
        volume_type           = "gp3"
        iops                  = 3000
#            throughput            = 150
        encrypted             = true
        kms_key_id            = aws_kms_key.ebs.arn
        delete_on_termination = true
      }
    }
  }

  # Determines whether the bootstrap configurations are populated within the user data template.
  # Only valid when using a custom AMI via ami_id
  enable_bootstrap_user_data = true

  # default AL2 admin container does not support cgroupv2 so disable it for now!
  bootstrap_extra_args = <<-EOT
  # extra args added
  [settings.kernel]
  lockdown = "integrity"

  #[settings.boot]
  #reboot-to-reconcile = true

  [settings.boot.kernel-parameters]
  ps1 = "1"

  #[settings.boot.init-parameters]
  #"systemd.unified_cgroup_hierarchy " = "1"

  [settings.host-containers.admin]
  enabled = true

  [settings.kubernetes.node-labels]
  "label1" = "waltju1"
  "label2" = "waltju2"
  EOT

  # MNG node role
  create_iam_role = false
  iam_role_arn    = aws_iam_role.eks-worker.arn

  # By default, the module creates a launch template to ensure tags are propagated to instances, etc.,
  # so we need to disable it to use the default template provided by the AWS EKS managed node group service
  # without this MNG will not join cluster
  create_launch_template = true
  launch_template_name   = ""

  ami_type = "BOTTLEROCKET_x86_64"
  platform = "bottlerocket"
  ami_id   = data.aws_ami.bottlerocket_ami.id
}

**Comment by youwalther65** [comment]
Closed by mistake

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.31.0 :tada:
